skimage/color/colorconv.py

```diff
@@ -155,34 +162,34 @@ def _prepare_rgba_array(arr):
     """
     arr = np.asanyarray(arr)
 
-    if arr.ndim not in [3, 4] or arr.shape[-1] != 4:
-        msg = ("the input array must have a shape == (.., ..,[ ..,] 4)), "
+    if arr.ndim < 2 or arr.shape[-1] != 4:
```

Comment:
this should probably be combined with the above, with the `n_channels` parameter added to switch between `arr.shape[-1] != n_channels`

Reply:
I was even more hesitating to remove this function since it is only used in rgba2rgb